tests/end_to_end/EvmPrecompiles/test_evm_precompiles.py

```diff
@@ -0,0 +1,134 @@
+import pytest
+import pytest_asyncio
+from ethereum.base_types import U256, Uint
+from ethereum.cancun.vm.exceptions import OutOfGasError
+from ethereum.crypto.alt_bn128 import ALT_BN128_PRIME, BNF, BNP
+from hypothesis import given, settings
+from hypothesis.strategies import integers
+
+from kakarot_scripts.utils.kakarot import deploy
+
+
+@pytest_asyncio.fixture(scope="package")
+async def evm_precompiles(owner):
+    return await deploy(
+        "EvmPrecompiles",
+        "EvmPrecompiles",
+        caller_eoa=owner.starknet_contract,
+    )
+
+
+def ref_alt_bn128_add(x0, y0, x1, y1):
+    x0_value = U256.from_signed(x0)
+    y0_value = U256.from_signed(y0)
+    x1_value = U256.from_signed(x1)
+    y1_value = U256.from_signed(y1)
+
+    for i in (x0_value, y0_value, x1_value, y1_value):
+        if i >= ALT_BN128_PRIME:
+            raise OutOfGasError
+
+    try:
+        p0 = BNP(BNF(x0_value), BNF(y0_value))
+        p1 = BNP(BNF(x1_value), BNF(y1_value))
+    except ValueError:
+        raise OutOfGasError from None
```

Comment:
let's just remove the try/except then and keep the valueerror

Reply:
may be easier to read the test if the ref implementation has the same signature, meaning it returns (bool x, y) and don't raise